distribution.md

```diff
@@ -43,29 +44,219 @@ Each buildpack layer blob MUST contain a single buildpack at the following file
+  "id": "<buildpack ID (required)>",
+  "name": "<buildpack name (optional)>",
+  "version": "<buildpack version (required)>",
+  "homepage": "<buildpack home page (optional)>",
+}
+```
+
+##### `io.buildpacks.buildpack.layers` (JSON)
+
+```json
+{
+  # schema of a meta-buildpack
+  "<buildpack ID (required)>": {
+    "<buildpack version (required)>": {
+      "api": "<buildpack API (required)>",
+      "order": [
+        {
+          "group": [
+            {
+              "id": "<inner buildpack ID (required)>",
+              "version": "<inner buildpack version (required)>"
+            }
+          ]
+        }
+      ],
+      "layerDiffID": "<diff ID of buildpack layer (required)>",
+      "homepage": "<buildpack homepage (optional)>",
+      "name": "<buildpack name (optional)>"
+    }
+  },
+  # schema of a regular buildpack
+  "<buildpack ID (required)>": {
+    "<buildpack version (required)>": {
+      "api": "<buildpack API (required)>",
+      "layerDiffID": "<diff ID of buildpack layer (required)>",
+      "homepage": "<buildpack homepage (optional)>",
+      "name": "<buildpack name (optional)>"
     }
-  ]
+  }
 }
 ```
 
+For each buildpack layer, the buildpack ID and the buildpack version MUST be provided in `io.buildpacks.buildpack.layers`.
+
 The buildpack ID and version MUST match a buildpack provided by a layer blob.
 
-For a buildpackage to be valid, each `buildpack.toml` describing a buildpack implementation MUST have all listed stacks.
+For a buildpackage to be valid, each `buildpack.toml` describing a buildpack implementation MUST have all listed targets.
+
+### Builder
+
+The following specifies the artifact format for a buildpacks builder.
+
+A builder is an OCI Image that provides a distributable build environment.
+
+A platform supporting builders SHOULD allow users to configure the build environment for a provided builder.
+
+#### General Requirements
+
+The builder image MUST contain an implementation of the [lifecycle](#lifecycle), and a [build-time](#build-image) environment, and MAY contain [buildpacks](#buildpackage). Platforms SHOULD use builders to ease the build process.
+
+#### Filesystem
+
+A builder MUST have the following directories/files:
+
+* `/cnb/order.toml` &rarr; As defined in the [platform specification](https://github.com/buildpacks/spec/blob/main/platform.md#ordertoml-toml).
+
+#### Environment Variables
+
+A builder MUST be an extension of a Build Image, and MUST retain all the specified environment variables set on the original build image, as specified in the Build Image specifications.
+
+#### Labels
+
+A builder MUST be an extension of a Build Image, and MUST retain all the specified Labels set on the original build image, as specified in the Build Image specifications.
+
+A builder image MUST contain an implementation of the [lifecycle](#lifecycle), and MUST retain all the specified Labels set on the original Lifecycle image, as specified in the lifecycle distribution specifications.
```

Comment:
```suggestion
A builder MUST contain an implementation of the [lifecycle](#lifecycle), and MUST retain all the specified Labels set on the original Lifecycle image, as specified in the lifecycle distribution specifications.
```

Should we refer to it as "builder image" everywhere, or just "builder"?

Reply:
makes sense,  i'll normalize to "builder"